packages/react-ui/src/features/templates/components/share-template.tsx

```diff
@@ -43,61 +38,66 @@ const ShareTemplate: React.FC<{ templateId: string }> = ({ templateId }) => {
-                        key={index}
-                      />
-                    ))}
-                  </div>
-                </div>
-                {data.description && (
-                  <>
-                    <Separator className="my-2" />
-                    <div className="flex flex-col w-full justify-start items-start">
-                      <span className="font-semibold">Description</span>
-                      <span>{data.description}</span>
-                    </div>
-                  </>
-                )}
-              </div>
-              <div className="ml-auto">
-                <Button
-                  variant={'secondary'}
-                  className="mr-2"
-                  onClick={() => navigate('/flows')}
-                >
-                  Cancel
-                </Button>
-                <Button loading={isPending} onClick={() => mutate()}>
-                  Confirm
-                </Button>
+      <>
+        <CardHeader>
+          <span className="font-semibold">{template.name}</span>
+          <Separator />
+        </CardHeader>
+        <CardContent>
+          <div className="flex flex-col gap-2">
+            <div className="flex flex-col gap-2 items-center justify-between mb-4">
+              <div className="flex flex-row w-full justify-between items-center">
+                <span>Steps in this flow</span>
+                <PieceIconList
+                  trigger={template.template.trigger}
+                  maxNumberOfIconsToShow={5}
+                ></PieceIconList>
               </div>
+              {template.description && (
+                <>
+                  <Separator className="my-2" />
+                  <div className="flex flex-col w-full justify-start items-start">
+                    <span className="font-semibold">Description</span>
+                    <span>{template.description}</span>
+                  </div>
+                </>
+              )}
+            </div>
+            <div className="ml-auto">
+              <Button
+                variant={'secondary'}
+                className="mr-2"
+                onClick={() => navigate('/flows')}
+              >
+                Cancel
+              </Button>
+              <Button loading={isPending} onClick={() => mutate()}>
+                Confirm
+              </Button>
             </div>
-          </CardContent>
-        </>
-      )}
+          </div>
+        </CardContent>
+      </>
     </Card>
   );
 };
 
+const ShareTemplate: React.FC<{ templateId: string }> = ({ templateId }) => {
+  const { data } = useQuery({
+    queryKey: ['template', templateId],
+    queryFn: () => templatesApi.getTemplate(templateId),
+    staleTime: 0,
+  });
+  if (!data) {
+    return (
+      <div className="w-full h-full flex items-center justify-center">
+        <LoadingSpinner size={50}></LoadingSpinner>
+      </div>
+    );
+  }
+  return <TemplateViewer template={data}></TemplateViewer>;
```

Comment:
Why this can't be single component where FlowTemplate handle the loading as well? I don't see where TemplateViewer can be reused, If it has to be resuable later then It's better to seperate in different component.

Reply:
Template viewer uses hooks, and those hooks depend on the template, making sure the template is defined before the component is rendered makes it easier to handle those hooks instead of checking for the template being defined or not.